pkg/scheduler/util/non_zero.go

```diff
@@ -34,45 +36,103 @@ const (
 	DefaultMilliCPURequest int64 = 100 // 0.1 core
 	// DefaultMemoryRequest defines default memory request size.
 	DefaultMemoryRequest int64 = 200 * 1024 * 1024 // 200 MB
+	// DefaultEphemeralStorage defines default empheral storage request size.
+	DefaultEphemeralStorage int64 = 0 // 0 Gi
 )
 
 // GetNonzeroRequests returns the default cpu and memory resource request if none is found or
 // what is provided on the request.
-func GetNonzeroRequests(requests *v1.ResourceList) (int64, int64) {
-	return GetNonzeroRequestForResource(v1.ResourceCPU, requests),
-		GetNonzeroRequestForResource(v1.ResourceMemory, requests)
+func GetNonzeroRequests(pod *v1.Pod) (int64, int64) {
+	return GetNonzeroRequestForResource(pod, v1.ResourceCPU),
+		GetNonzeroRequestForResource(pod, v1.ResourceMemory)
 }
 
 // GetNonzeroRequestForResource returns the default resource request if none is found or
 // what is provided on the request.
-func GetNonzeroRequestForResource(resource v1.ResourceName, requests *v1.ResourceList) int64 {
```

Comment:
you could leave this function, make it unexported and call it from the loop that iterates over pods.

Reply:
revised